### PR TITLE
WIP: fix(vagrant): Fix Vagrantfile.

### PR DESCRIPTION
**Pull request by robertvolkmann**
Changes
---------
* Removed symlinks.
* Switch to debian/jessie64.
* Added missing test dependency.
* Added script to configure vagrant for development.

How to test
-------------
```
$ vagrant up
$ vagrant ssh
$ cd /vagrant
$ ./utils/prepare-vagrant-dev.sh
$ PGHOST=localhost make test
```

### CODE REVIEW

**Comment by robertvolkmann** [comment]
Actually the following test cases for ununpack are failing.
```
There were 5 failures:

1) cliParamsTest4Ununpack::testNormalAr
Failed asserting that file "/vagrant/src/ununpack/agent_tests/Functional/test_result/test.deb.dir/control.tar.gz.dir/md5sums" exists.

/vagrant/src/ununpack/agent_tests/Functional/cliParamsTest4UnunpackNormal.php:215

2) cliParamsTest4Ununpack::testNormalCatMsi
Failed asserting that file "/vagrant/src/ununpack/agent_tests/Functional/test_result/test.msi.dir/ununpack" exists.

/vagrant/src/ununpack/agent_tests/Functional/cliParamsTest4UnunpackNormal.php:265

3) cliParamsTest4Ununpack::testNormalCompressedFile
Failed asserting that file "/vagrant/src/ununpack/agent_tests/Functional/test_result/fossI16L335U29.tar.bz2.dir/fossology/README" exists.

/vagrant/src/ununpack/agent_tests/Functional/cliParamsTest4UnunpackNormal.php:312

4) cliParamsTest4Ununpack::testNormalTarball
Failed asserting that file "/vagrant/src/ununpack/agent_tests/Functional/test_result/test.tar.Z.dir/dir1/ununpack" exists.

/vagrant/src/ununpack/agent_tests/Functional/cliParamsTest4UnunpackNormal.php:325

5) cliParamsTest4Ununpack::testNormalFsImage
Failed asserting that file "/vagrant/src/ununpack/agent_tests/Functional/test_result/ext3file.fs.dir/testtwo.zip.dir/test.zip.dir/ununpack" exists.

/vagrant/src/ununpack/agent_tests/Functional/cliParamsTest4UnunpackNormal.php:386

FAILURES!
```

**Comment by robertvolkmann** [comment]
Thx to @GMishx, rebasing after the merge of #1092 fixed the failed ununpack tests.

File: utils/prepare-vagrant-dev.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+sudo apt-get install -q -y libcppunit-dev \
+          libspreadsheet-writeexcel-perl libdbd-sqlite3-perl \
+          openjdk-7-jre-headless
+
+./utils/prepare-test -afty
```
Comment (maxhbr) [inline utils/prepare-vagrant-dev.sh]: If one calls this script this way, it breaks the current FOSSology instance without warning or asking. Please remove the `-a` flag.

File: utils/prepare-vagrant-dev.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+./install/scripts/install-ninka.sh
+./install/scripts/install-spdx-tools.sh
+
+composer install --working-dir=src
```
Comment (maxhbr) [inline utils/prepare-vagrant-dev.sh]: This script does not work if someone calls
```
cd utils
./prepare-vagrant-dev.sh
```

File: Vagrantfile [hunk 3]
```diff
@@ -20,37 +20,27 @@ fi
...
-
 date > /etc/vagrant.provisioned
 
 echo "lets go!"
 cd /vagrant
 
 ./utils/fo-installdeps -e -y
 
-make CFLAGS=-I/usr/include/glib-2.0
+make all
```
Comment (maxhbr) [inline Vagrantfile]: What is the difference between `make` and `make all`?

File: Vagrantfile [hunk 4]
```diff
@@ -20,37 +20,27 @@ fi
...
 sudo make install
+# fix ownership
+sudo chown -R vagrant:vagrant /vagrant
```
Comment (maxhbr) [inline Vagrantfile]: This modifies the permissions on the mounted folder, i.e. might cause permission problems for the user on the host system?

This is done after calling `make all`, so if there were permission problems this is already to late?

**Comment by maxhbr** [comment]
Code looks good, but this changes the way, in which the `/vagrant` dir is mounted, from `vboxsf` to `rsync`, i.e. changes on the host are not directly visible in the virtual machine, unless e.g. `vagrant rsync` is called. 

For automatic synchroniation one can use the command `vagrant rsync-auto`.

**Comment by robertvolkmann** [comment]
Sry for this undocumented breaking change, but since this PR switched the base box from trusty to jessie we have already a breaking change.

**Comment by maxhbr** [comment]
maybe also add
```diff
diff --git a/Vagrantfile b/Vagrantfile
index cd01cace4..54f07259d 100755
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -52,6 +52,7 @@ SCRIPT
 
 Vagrant.configure("2") do |config|
   config.vm.box = "debian/jessie64"
+  config.vm.post_up_message = "Welcome to FOSSology!"
 
   config.vm.provider "virtualbox" do |vbox|
     vbox.customize ["modifyvm", :id, "--memory", "4096"]
```

to replace the debian `post_up_message`:
```
==> default: Vanilla Debian box. See https://atlas.hashicorp.com/debian/ for help and bug reports
```

**Comment by maxhbr** [comment]
I would suggest to do something like the following
```diff
diff --git a/Vagrantfile b/Vagrantfile
index cd01cace4..81ebedc0e 100755
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -3,6 +3,16 @@
 
 # Copyright Siemens AG, 2014
 # SPDX-License-Identifier:  GPL-2.0 LGPL-2.1
+$welcomeMsg = <<WELCOME
+Use your FOSSology at http://localhost:8081/repo/
+  user: fossy , password: fossy
+
+Or do a 'vagrant ssh' and look at /fossology for your source tree.
+
+To update the sources within the vagrant box you have to either
+- call 'vagrant rsync' maually or
+- call 'vagrant rsync-auto' to tell vagrant, to automatically sync the changes
+WELCOME
 
 $add_proxy_settings = <<PROXYSCRIPT
 # get gateway of running VM and set proxy info within global profile
@@ -29,7 +39,7 @@ sudo apt-get update -qq -y
 date > /etc/vagrant.provisioned
 
 echo "lets go!"
-cd /vagrant
+cd /fossology
 
 ./utils/fo-installdeps -e -y
 
@@ -38,20 +48,17 @@ sudo make install
 
 sudo /usr/local/lib/fossology/fo-postinstall
 
-sudo cp /vagrant/install/src-install-apache-example.conf /etc/apache2/conf-enabled/fossology.conf
+sudo cp /fossology/install/src-install-apache-example.conf /etc/apache2/conf-enabled/fossology.conf
 
 # increase upload size
-sudo /vagrant/install/scripts/php-conf-fix.sh --overwrite
+sudo /fossology/install/scripts/php-conf-fix.sh --overwrite
 
 sudo /etc/init.d/apache2 restart
-
-echo "use your FOSSology at http://localhost:8081/repo/"
-echo " user: fossy , password: fossy"
-echo "or do a vagrant ssh and look at /vagrant for your source tree"
 SCRIPT
 
 Vagrant.configure("2") do |config|
   config.vm.box = "debian/jessie64"
+  config.vm.post_up_message = $welcomeMsg
 
   config.vm.provider "virtualbox" do |vbox|
     vbox.customize ["modifyvm", :id, "--memory", "4096"]
@@ -70,6 +77,9 @@ Vagrant.configure("2") do |config|
     end
   end
 
+  config.vm.synced_folder ".", "/fossology",
+                          type: "rsync"
+
   # call the script
   config.vm.provision "shell" do |shell|
     shell.inline = $build_and_test
```

**Comment by maxhbr** [comment]
@mcjaeger this updates the base box of the Vagrant box from `"trusty64"` to `"debian/jessie64"` (which matches the base of our Dockerfiles. But, since the jessie image comes without installed vagrant guest modules, this implies that the source code has to be mounted via "rsync" instead of "vboxsf" which chages the way, how one can update the source in the box via the host.

Previously all changes were instantaniously available in the VM, since the folder was mounted directly.

Rsync does this in a more indirect way. E.g. to update the sources one has to run `vagrant rsync` on the host. But there is also the possibility to automatically sync the folder via `vagrant rsync-auto`.

If one starts the vagrant box, the new way of how to update the source is displayed as welcome message.

Maybe you want to test this PR and say, whether the changes are good for you.

File: Vagrantfile [hunk 5]
```diff
@@ -3,6 +3,16 @@
 
 # Copyright Siemens AG, 2014
 # SPDX-License-Identifier:  GPL-2.0 LGPL-2.1
+$welcomeMsg = <<WELCOME
```
Comment (robertvolkmann) [inline Vagrantfile]: Ruby style guides advises snake_case.
See https://github.com/rubocop-hq/ruby-style-guide#naming

File: Vagrantfile [hunk 6]
```diff
@@ -70,6 +77,9 @@ Vagrant.configure("2") do |config|
     end
   end
 
+  config.vm.synced_folder ".", "/fossology",
```
Comment (robertvolkmann) [inline Vagrantfile]: You should deactivate the sync to `/vagrant` with:
`config.vm.synced_folder ".", "/vagrant", disabled: true`

Comment (maxhbr) [reply Vagrantfile]: That is annoying, but I think it is still worth it, to call the folder `"/fossology"`

**Comment by robertvolkmann** [comment]
After running `vagrant rsync` all changes in `/fossology` are removed because the [default arguments](https://www.vagrantup.com/docs/synced-folders/rsync.html#options) for the rsync contains `--delete`.

Hence we adjust the rsync arguments or we try the vagrant plugin [vbguest](https://github.com/dotless-de/vagrant-vbguest).

**Comment by robertvolkmann** [comment]
@maxhbr @mcjaeger I create the PR #1207 for a better approach to fix the Vagrantfile. This PR can be closed without merging.

**Pull request closed without merging**